Drivers/Src/MotorController.c

```diff
@@ -0,0 +1,71 @@
+#include "MotorController.h"
+
+
+#define MOTOR_DRIVE 0x221
+
+#define SUCCESS     1
+#define FAILURE     0
+
+
+
+void MotorController_Init(){
+    BSP_CAN_Init(CAN_2);
+}
+
+
+void MotorController_Drive(uint32_t newVelocity, uint32_t motorCurrent){
+    uint8_t data[8] = {0};
+    int index = 0;
+    while(index < 4){
+        data[index] = (newVelocity >> (8 * (4-index-1))) & 0xFF;    //split inputs into bytes
+        index++;
+        
+    }
+    int i = 0;
+    while(index < 8){
+        data[index] = (motorCurrent >> (8 * (4-i-1))) & 0xFF;
+        index++;
+        i++;
+    }
+    
+    BSP_CAN_Write(CAN_2, MOTOR_DRIVE, data, 8);
+}
+
+bool MotorController_Read(CANbuff *message, uint32_t expectedID){
```

Comment:
I would take out the expected ID here. We want to simply get whatever message is being sent on the Motor Controller CAN bus and pack it into a struct. That way, we can put it into our FIFO eventually and forward the messages to Data Acquisition.